trieste/bayesian_optimizer.py

```diff
@@ -634,18 +645,12 @@ def optimize(
         if not datasets:
             raise ValueError("dicts of datasets and models must be populated.")
 
-        if fit_model not in ["all", "all_but_init", "never"]:
-            raise ValueError(
-                f"Expecting fit_model to be 'all', 'all_but_init' or 'never', "
-                f"but received {fit_model!r}"
-            )
-
-        if (fit_model in ["all", "all_but_init"]) and isinstance(acquisition_rule, TURBO):
+        if fit_model and isinstance(acquisition_rule, TURBO):
```

Comment:
why not having a subclass of `AcquisitionRule` like `AcquisitionRuleWithLocalModels`  that we check against, rather than specific acq rules?

Reply:
Probably makes sense if we end up with another example of this. Best not to preemptively overengineer.